xapian-core/include/xapian/queryparser.h

```diff
@@ -234,6 +234,79 @@ class XAPIAN_VISIBILITY_DEFAULT RangeProcessor
     }
 };
 
+// Handles a file size range
+
+class XAPIAN_VISIBILITY_DEFAULT FileSizeRangeProcessor : public RangeProcessor {
+  public:
+    /** Constructor.
+     *
+     *  @param slot_    The value slot number to query.
+     *
+     *  @param str_     A string to look for to recognise values as belonging
+     *                  to this file-size range.
+     *
+     *  @param flags_   Zero or more of the following flags, combined with
+     *          bitwise-or:
+     *           * Xapian::RP_SUFFIX - require @a str_ as a suffix
+     *             instead of a prefix.
+     *           * Xapian::RP_REPEATED - optionally allow @a str_
+     *             on both ends of the range - e.g. $1..$10 or
+     *             5m..50m.  By default a prefix is only checked for on
+     *             the start (e.g. date:1/1/1980..31/12/1989), and a
+     *             suffix only on the end (e.g. 2..12kg).
+     *
+     *  The string supplied in str_ is used by @a operator() to decide whether
+     *  the pair of strings supplied to it constitute a valid range.  If
+     *  prefix_ is true, the first value in a range must begin with str_ (and
```

Comment:
I don't see a sensible use for being able to specify a suffix - you could have some magic suffix indicate this is a size range, but (say) `1..100K^` just seems weird.

And repeating the prefix/suffix also doesn't seem useful here.  That's really meant for "units", e.g. `1..10kg` vs `1kg..10kg`; `$1.100` vs `$1..$100`, but I don't see that being helpful here.

However, I don't think we want to make it impossible to specify these as people may find uses (and there may be more flags in the future, and some of those may be useful here even if the current flags aren't).  So discussing in the doccomment seems best to me.